### PR TITLE
#19646 Improvment heap scape consumption generating a bundle

### PR DESCRIPTION
**Pull request by freddyucv**
Create a new waitForAll method to improve the before implementation, now it not need to keep in memory a Future Collection.

https://github.com/dotCMS/core/pull/20345/files#diff-eb86bd7229e6b229dd5fbcab201c7948ceeb9b17f90a6bc501e71b4e7dd9f2a5R740

Also create the getSessionIfOpened to use in the evict method, this new 'getSessionIfOpened' just return a session if exists one already opened and it never open a new session (this change avoid a leak connection error that we was gottinh before)

https://github.com/dotCMS/core/pull/20345/files#diff-97d93998d67c9cb195bdae899f747d17ddcfcc3d3a71392dfd74b2df40648f4aR721

Finally just in case I closed the Entry in the finally section

https://github.com/dotCMS/core/pull/20345/files#diff-ad30c1685bed7089a0808e5fb617e74e6f11f5e2c73ed0a7fb395293997caa8bR123

All this methods are really util and it are used here:

https://github.com/dotCMS/enterprise/pull/796

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/publishing/output/TarGzipBundleOutput.java [hunk 1]
```diff
@@ -116,9 +116,12 @@ public void close() throws IOException {
 
         private void putEntry(byte[] bytes, TarArchiveEntry tarArchiveEntry) throws IOException {
```
Comment (wezell) [inline dotCMS/src/main/java/com/dotcms/publishing/output/TarGzipBundleOutput.java:117]: Well, there's your culprit, right?  Are reading whole files into memory as byte[]?

File: dotCMS/src/main/java/com/dotcms/concurrent/DotConcurrentFactory.java [hunk 2]
```diff
@@ -736,6 +736,13 @@ public void waitForAll(long timeout, TimeUnit unit) throws ExecutionException {
             }
         }
 
+        @Override
+        public void waitForAll() throws ExecutionException {
+            while(executorService.isTerminated()) {
+                waitForAll(10, TimeUnit.MINUTES);
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/concurrent/DotConcurrentFactory.java:742]: 10 minutes is not too long?

Comment (freddyucv) [reply dotCMS/src/main/java/com/dotcms/concurrent/DotConcurrentFactory.java:742]: it is 10 minutes to check again, if the task finish before it is awake

File: dotCMS/dependencies.gradle [hunk 3]
```diff
@@ -3,7 +3,8 @@ dependencies {
 
     def eeType = dotcmsReleaseVersion.equals("master") ? '-SNAPSHOT' : ''
 
-    compile group: 'com.dotcms.enterprise', name: 'ee', version: dotcmsReleaseVersion + eeType, changing: true
+    compile group: 'com.dotcms.enterprise', name: 'ee', version: 'issue-19646-improvment-memory-space-consumption-SNAPSHOT', changing: true
```
Comment (dsilvam) [inline dotCMS/dependencies.gradle:6]: restore this value

**Review by dsilvam — APPROVED** [review]

**Review by victoralfaro-dotcms — APPROVED** [review]
LGTM

**Pull request closed without merging**